go-controller/pkg/ovn/controller/services/svc_template_var.go

```diff
@@ -215,3 +210,60 @@ func getTemplatesFromRulesTargets(rules []LBRule) TemplateMap {
 	}
 	return templates
 }
+
+// NodeIPTemplates maintains templates variables for many IP addresses per node,
+// creating them in the form NODEIP_IPv4_0, NODEIP_IPv4_1, NODEIP_IPv4_2, ...
+// if and when they are needed.
+type NodeIPsTemplates struct {
+	ipFamily  corev1.IPFamily
+	templates []*Template
+}
+
+func NewNodeIPsTemplates(ipFamily corev1.IPFamily) *NodeIPsTemplates {
+	return &NodeIPsTemplates{
+		ipFamily:  ipFamily,
+		templates: make([]*Template, 0),
+	}
+}
+
+// AddIP adds a template variable for the specified chassis and ip address.
+func (n *NodeIPsTemplates) AddIP(chassisID string, ip net.IP) {
+
+	var i int = -1
```

Comment:
hmm I see this part of code also changed since I previously did a review, not a fan of the starting with i = -1 just using that variable to get the total number of templates we have for the node a.k.a the nodeIPs..., if like @dceara mentions we can leverage the length and figure that out, let's do that...seems cleaner?